### PR TITLE
fix: Editor extensions: folding, numbered headings

### PR DESCRIPTION
**Pull request by richburdon**
- [x] Folding option
- [x] Numbered heading option
- [x] Misc
  - [x] NOTE: Remove default component props that are used as deps!
  - [x] Trim borders
  - [x] Fix chess layout
  - [x] Fix explorer plugin dark mode styles

## Editor options

https://github.com/user-attachments/assets/5e10439e-0901-4dd2-9583-b34d17ce9038

## Borders

Screenshot 2024-08-31 at 1 40 22 AM

### CODE REVIEW

File: packages/plugins/experimental/plugin-excalidraw/src/hooks/useStoreAdapter.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export const useStoreAdapter = (
     }
 
     const t = setTimeout(async () => {
-      await adapter.open(createDocAccessor(object, ['content']));
+      await adapter.open(createDocAccessor(object, ['canvas', 'content']));
```
Comment (wittjosiah) [inline packages/plugins/experimental/plugin-excalidraw/src/hooks/useStoreAdapter.ts:31]:
```suggestion
      await adapter.open(createDocAccessor(object.canvas, ['content']));
```
is there a difference between these two? if the canvas isn't loaded will creating a deeply nested doc accessor load the object? shouldn't merge this without the corresponding migration?

File: packages/plugins/plugin-chess/src/components/index.ts [hunk 2]
```diff
@@ -5,5 +5,4 @@
 import React from 'react';
 
 // Lazily load components for content surfaces.
-export const ChessMain = React.lazy(() => import('./ChessMain'));
-export const ChessArticle = React.lazy(() => import('./ChessArticle'));
+export const ChessWrapper = React.lazy(() => import('./ChessWrapper'));
```
Comment (wittjosiah) [inline packages/plugins/plugin-chess/src/components/index.ts:8]: ChessContainer? if aligning with other plugins

**Review by wittjosiah — COMMENTED** [review]
love to see folding making it's way into the editor (looking forward to it in lists next 😉). Not sure if you ran into a technical limitation here or not, but I think it would be more useable if the collapse button was right next to the content instead of over off to the side

**Comment by richburdon** [comment]
> love to see folding making it's way into the editor (looking forward to it in lists next 😉). Not sure if you ran into a technical limitation here or not, but I think it would be more useable if the collapse button was right next to the content instead of over off to the side

:). i'm not sure how to do that given the content margins. let's discuss with the team.

File: packages/plugins/experimental/plugin-excalidraw/src/hooks/useStoreAdapter.ts (hunk 1 above)
Comment (richburdon) [reply packages/plugins/experimental/plugin-excalidraw/src/hooks/useStoreAdapter.ts:31]: ok, i cleaned this up before this commit -- and i think i had confused myself. no, this is correct. and a migration won't be needed. but it doesn't work in the storybook.

File: packages/plugins/plugin-chess/src/components/index.ts (hunk 2 above)
Comment (richburdon) [reply packages/plugins/plugin-chess/src/components/index.ts:8]: ah yes, that was the intention. thanks.

**Pull request merged**